src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcOfferDao.java

```diff
@@ -0,0 +1,138 @@
+
+    public JdbcOfferDao(HikariDataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public List<Offer> findManyByUniqueVinylId(long uniqueVinylId) {
+        List<Offer> offers = new ArrayList<>();
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement findVinylByIdStatement = connection.prepareStatement(SELECT_MANY_BY_UNIQUE_VINYL_ID)) {
+            findVinylByIdStatement.setLong(1, uniqueVinylId);
+            logger.debug(PREPARED_STATEMENT_MESSAGE, findVinylByIdStatement);
+            try (ResultSet resultSet = findVinylByIdStatement.executeQuery()) {
+                boolean isResultSetEmpty = true;
+                while (resultSet.next()) {
+                    isResultSetEmpty = false;
+                    Offer offer = rowMapper.mapRow(resultSet);
+                    offers.add(offer);
+                }
+                if (isResultSetEmpty) {
+                    RuntimeException e = new RuntimeException();
+                    logger.error(NO_ROW_BY_ERROR_MESSAGE, OBJECT_NAME, "uniqueVinylId", "uniqueVinylId", uniqueVinylId, e);
+                    throw e;
+                }
+            }
+        } catch (SQLException e) {
+            logger.error(FINDING_BY_ERROR_MESSAGE, OBJECT_NAME, "uniqueVinylId", "uniqueVinylId", uniqueVinylId, e);
+            throw new RuntimeException(e);
+        }
+        logger.debug(RESULT_IS_MESSAGE, LIST_NAME, LIST_NAME, offers);
+        return offers;
+    }
+
+    @Override
+    public List<Offer> updateUniqueVinylsRewriteAll(List<UniqueVinyl> uniqueVinyls, List<Offer> offers) {
+        if (uniqueVinyls == null) {
+            RuntimeException e = new NullPointerException("List of unique vinyls is null");
+            logger.error("List of unique vinyls is null", e);
+            throw e;
+        }
+        if (offers == null) {
+            RuntimeException e = new NullPointerException("List of offers is null");
+            logger.error("List of offers is null", e);
+            throw e;
+        }
+        if (uniqueVinyls.isEmpty()) {
+            RuntimeException e = new IllegalArgumentException("List of unique vinyls is empty");
+            logger.error("List of unique vinyls is empty", e);
+            throw e;
+        }
+        if (offers.isEmpty()) {
+            RuntimeException e = new IllegalArgumentException("List of offers is empty");
+            logger.error("List of offers is empty", e);
+            throw e;
+        }
+        List<Offer> notAddedOffers = new ArrayList<>(offers);
+        try(Connection connection = dataSource.getConnection();
+            Statement prepareTables = connection.createStatement();
+            PreparedStatement upsertUniqueVinyls = connection.prepareStatement(UPSERT_UNIQUE_VINYLS);
+            PreparedStatement insertValidOffers = connection.prepareStatement(INSERT_VALID)) {
+            connection.setAutoCommit(false);
```

Comment:
Should we return connection to the pool with the same state it was taken from?
Otherwise some next transaction without explicit commit can be impacted...
If so, autoCommit property should be set to the previous value

Reply:
Thank you for pointing that out!